payjoin/src/uri.rs

```diff
@@ -274,51 +359,154 @@ enum InternalPjParseError {
     UnsecureEndpoint,
 }
 
+/// Builder for `bip21::Uri` with `PayjoinParams`.
+///
+/// `amount` parameter is optional and can be changed with `[PjUriBuilder::amount]`.
+///
+/// `ohttp_config` parameter is required only for v2 payjoin.
+pub struct PjUriBuilder {
+    /// Address you want to receive funds to.
+    ///
+    /// Must be a valid bitcoin address.
+    address: Address,
+    /// Payjoing endpoint url listening for payjoin requests.
+    ///
+    /// Must be a valid url that can be parsed
+    /// with `[Payjoin::Url::parse]`.
+    pj_endpoint: Url,
+    /// Amount you want to receive.
+    ///
+    /// If `None` the amount will be left unspecified.
+    amount: Option<Amount>,
+    /// Message
+    message: Option<String>,
+    /// Label
+    pub(crate) label: Option<String>,
+    #[cfg(feature = "v2")]
+    /// Config for ohttp.
+    ///
+    /// `[PjUriBuilder::decode_ohttp_config]` can be used to convert
+    /// base64 string to `ohttp::KeyConfig`.
+    ///
+    /// Required only for v2 payjoin.
+    ohttp_config: ohttp::KeyConfig,
+}
+
+impl PjUriBuilder {
+    /// Create a new `PjUriBuilder` with required parameters.
+    pub fn new(
+        address: Address,
+        pj_endpoint: Url,
+        amount: Option<Amount>,
+        message: Option<String>,
+        label: Option<String>,
+        #[cfg(feature = "v2")] ohttp_config: ohttp::KeyConfig,
```

Comment:
If we're using a builder these could all be modified with impl functions rather than included in `new()`